src/mlpack/bindings/markdown/binding_info.hpp

```diff
@@ -47,8 +79,20 @@ class BindingInfo
   //! Get the singleton.
   static BindingInfo& GetSingleton();
 
-  //! Internally-held map for mapping a binding name to a ProgramDoc name.
-  std::unordered_map<std::string, util::ProgramDoc> map;
+  //! Internally-held map for mapping a binding name to a ProgramName name.
+  std::unordered_map<std::string, util::ProgramName> mapProgramName;
+
+  //! Internally-held map for mapping a binding name to a ShortDescription name.
+  std::unordered_map<std::string, util::ShortDescription> mapShortDescription;
+
+  //! Internally-held map for mapping a binding name to a LongDescription name.
+  std::unordered_map<std::string, util::LongDescription> mapLongDescription;
+
+  //! Internally-held map for mapping a binding name to a Example name.
+  std::unordered_map<std::string, std::vector<util::Example>> mapExample;
+
+  //! Internally-held map for mapping a binding name to a SeeAlso name.
+  std::unordered_map<std::string, std::vector<util::SeeAlso>> mapSeeAlso;
```

Comment:
It might be simpler to collect all of the information relevant to binding information in one struct, like `util::BindingDetails` or something like that.  That would also make it easier to add new things (if we needed to do that in the future), and here in `markdown::BindingInfo` you'd only need a function like `GetBindingDetails()` and `RegisterBindingDetails()`.  Let me know if I overlooked something there; just an idea. :+1:

Reply:
In some last commit, I have tried to collect all the information in one `struct`, Please let me know whether my approach is correct or not.

Reply:
Yeah, this looks like it works just fine.  Nice work! :+1: